jgrapht-core/src/main/java/org/jgrapht/alg/interfaces/KShortestPathAlgorithm.java

```diff
@@ -39,7 +39,36 @@
      * @param source the source vertex
      * @param sink the target vertex
      * @return a list of shortest paths
+     * @deprecated In favor of providing k as a parameter.
      */
-    List<GraphPath<V, E>> getPaths(V source, V sink);
+    @Deprecated
+    default List<GraphPath<V, E>> getPaths(V source, V sink) { 
+        return getPaths(source, sink, Integer.MAX_VALUE);
+    }
+    
+    /**
+     * Get a list of k-shortest paths from a source vertex to a sink vertex. If no such paths exist
+     * this method returns an empty list.
+     * 
+     * @param source the source vertex
+     * @param sink the target vertex
+     * @param k the number of shortest paths to return
+     * @return a list of the k-shortest paths
+     */
+    default List<GraphPath<V, E>> getPaths(V source, V sink, int k) {
```

Comment:
I would propose to have only one method in this interface:
` List<GraphPath<V, E>> getPaths(V source, V sink, int k)`
The other method, `Iterator<GraphPath<V, E>> iterator(V source, V sink, int k)` seems rather weird: you would have an iterator for which you can only invoke the `next()` method `k  times. In this case, you might as well invoke `getPaths(source, sink,k).iterator();`

Reply:
The only point for the iterator would be to allow the algorithm to construct the paths incrementally and return them one at a time. I am not particularly in favor of one or the other approach.